ee/modules/110-istio/hooks/revisions_monitoring_test.go

```diff
@@ -315,7 +315,7 @@ spec:
 			Expect(string(f.LogrusOutput.Contents())).To(HaveLen(0))
 
 			m := f.MetricsCollector.CollectedMetrics()
-			Expect(m).To(HaveLen(7))
+			Expect(m).To(HaveLen(16))
```

Comment:
Completely unobvious what happened here and how I easily see the right number from the tests code

Reply:
I suggest splitting this big test case with 11 test cases, each for one pod case. In every case for a pod, you can check that it either adds to the metric or, importantly, that it is not added. 

For now, it is just hard to track what pods are not added. It took some time for me to see the "ignored" one which is the first, but is should be easily be read and not make people read through this. I still didn't figure out which else pod is missing here. I'd like not to do this kind of hand job. We can structure tests better and reduce boilerplate to track narrow cases instead of all-in-one assertion.

Reply:
There will be refactoring in https://github.com/deckhouse/deckhouse/issues/2182. So the tests will be refactored too.

Reply:
I'll take your word for it!